File: lab4/fact.py
import math
import numpy as np

"""
Calculate factorial
"""

def myfact(input_num):
    
    # check the input number if it's 0/negative first
    if input_num < 0:
        print("Sorry, factorial does not exist for negative numbers")
    elif input_num == 0:
        print("The factorial of 0 is 1")
    elif input_num>0:
        s = 1
        for k in range(1,input_num):
            s = s*(k+1)
        return s
    

# def main():
#     input_num = int(input("Enter a number to find its factorial: "))
#     fact_result = factorial(input_num)
#     print("The factorial of", input_num, "is", fact_result)

# if __name__ == "__main__":
#     main()

File: lab7/fact.c
#include <stdio.h>

/* c code calculating factorial of a number */

int main() {
    int num, factorial = 1;

    // take input integer number = num
    printf("Enter an integer: ");
    scanf("%d", &num);
    
    // check if num is negative
    if (num < 0)
    {
        printf("Factorial of a negative number doesn't exist.\n");
    }
    // check if num is zero
    else if (num == 0)
    {
        printf("Factorial of a zero is 1.\n");
    }
    // calculate factorial of positive num
    else if (num > 0)
    {
        for (int i = 1; i <= num; ++i)
        {
            factorial *= i;
        }
        printf("Factorial of %d = %d \n", num, factorial);
    }

    return 0;
}
